pytest_girder/setup.py

```diff
@@ -1,8 +1,29 @@
+import os
+
 from setuptools import find_packages, setup
 
+
+def prerelease_local_scheme(version):
+    """Return local scheme version unless building on master in CircleCI.
+    This function returns the local scheme version number
+    (e.g. 0.0.0.dev<N>+g<HASH>) unless building on CircleCI for a
+    pre-release in which case it ignores the hash and produces a
+    PEP440 compliant pre-release version number (e.g. 0.0.0.dev<N>).
+    """
+
+    from setuptools_scm.version import get_local_node_and_date
+
+    if 'CIRCLE_BRANCH' in os.environ and \
+       os.environ['CIRCLE_BRANCH'] == 'master':
+        return ''
+    else:
+        return get_local_node_and_date(version)
```

Comment:
Could we put this function as a util inside the girder package to avoid repeating it in every plugin?

Reply:
Not really without going to crazy town.  That would require adding girder to `setup_requires`.  And the import to the function would have to be guarded in a `try` block.

Reply:
Makes sense. Should we add this block to the plugin cookiecutter template?

Reply:
We could, but the cookiecutter doesn't include any CI/release infrastructure anyway.  I'm relatively indifferent.

Reply:
There is the possibility of a `setuptools-scm` plugin to deduplicate that function, but we can do that later.

Reply:
I don't think it belongs in the cookiecutter template because I don't think plugins should default to using setuptools-scm and circle-ci.

Reply:
Agreed. Let's not add it there.